kyo-core/shared/src/test/scala/kyo/QueueTest.scala

```diff
@@ -201,39 +203,161 @@ class QueueTest extends Test:
+            yield
+                assert(backlog.isDefined)
+                assert(offered.count(_.contains(true)) == backlog.get.size - size)
+                assert(isClosed)
+            )
+                .pipe(Choice.run).unit
+                .repeat(repeats)
+                .as(succeed)
+        }
 
-        def size()(using AllowUnsafe): Int =
-            if closed then throw new IllegalStateException("Queue is closed")
-            else elements.size
+        "concurrent close attempts" in run {
+            (for
+                size  <- Choice.get(Seq(0, 1, 2, 10, 100))
+                queue <- Queue.init[Int](size)
+                latch <- Latch.init(1)
+                offerFiber <- Async.run(
+                    latch.await.andThen(Async.parallel((1 to 100).map(i => Abort.run(queue.offer(i)))))
+                )
+                closeFiber <- Async.run(
+                    latch.await.andThen(Async.parallel((1 to 100).map(_ => queue.close)))
+                )
+                _        <- latch.release
+                offered  <- offerFiber.get
+                backlog  <- closeFiber.get
+                isClosed <- queue.closed
+            yield
+                assert(backlog.count(_.isDefined) == 1)
+                assert(backlog.flatMap(_.toList.flatten).size == offered.count(_.contains(true)))
+                assert(isClosed)
+            )
+                .pipe(Choice.run).unit
+                .repeat(repeats)
+                .as(succeed)
+        }
 
-        def full()(using AllowUnsafe): Boolean =
-            if closed then throw new IllegalStateException("Queue is closed")
-            else elements.size == capacity
+        "offer, poll and close" in run {
+            (for
+                size  <- Choice.get(Seq(0, 1, 2, 10, 100))
+                queue <- Queue.init[Int](size)
+                latch <- Latch.init(1)
+                offerFiber <- Async.run(
+                    latch.await.andThen(Async.parallel((1 to 100).map(i => Abort.run(queue.offer(i)))))
+                )
+                pollFiber <- Async.run(
+                    latch.await.andThen(Async.parallel((1 to 100).map(_ => Abort.run(queue.poll))))
+                )
+                closeFiber <- Async.run(latch.await.andThen(queue.close))
+                _          <- latch.release
+                offered    <- offerFiber.get
+                polled     <- pollFiber.get
+                backlog    <- closeFiber.get
+                isClosed   <- queue.closed
+            yield
+                assert(backlog.isDefined)
+                assert(offered.count(_.contains(true)) - polled.count(_.toMaybe.flatten.isDefined) == backlog.get.size)
+                assert(isClosed)
+            )
+                .pipe(Choice.run).unit
+                .repeat(repeats)
+                .as(succeed)
+        }
+    }
 
-    end TestUnsafeQueue
+    if Platform.isJVM then
+        "non-power-of-two capacity" - {
```

Comment:
jvm pending tests showing the behavior with queues rounding up the capacity to the next power of two